plugins/dbus_announce.c

```diff
@@ -0,0 +1,143 @@
+#include "system.h"
+
+#include <dbus/dbus.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <unistd.h>
+#include <rpm/rpmlog.h>
+#include <rpm/rpmts.h>
+#include <rpm/rpmdb.h>
+#include "lib/rpmplugin.h"
+
+struct dbus_announce_data {
+    DBusConnection * bus;
+    int logging;
+    int old_dboffset;
+};
+
+static rpmRC dbus_announce_init(rpmPlugin plugin, rpmts ts)
+{
+    struct dbus_announce_data * state = rcalloc(1, sizeof(*state));
+    rpmPluginSetData(plugin, state);
+    return RPMRC_OK;
+}
+
+static rpmRC open_dbus(rpmPlugin plugin, rpmts ts)
+{
+    struct stat st;
+    DBusError err;
+    int rc = 0;
+    struct dbus_announce_data * state = rpmPluginGetData(plugin);
+
+    /* Assume we are logging but... */
+    state->logging = 1;
+
+    /* ...don't log test transactions */
+    if (rpmtsFlags(ts) & (RPMTRANS_FLAG_TEST|RPMTRANS_FLAG_BUILD_PROBS))
+	state->logging = 0;
+
+    /* ...don't log chroot transactions */
+    if (!rstreq(rpmtsRootDir(ts), "/"))
+	state->logging = 0;
+
+    /* Don't open */
+    if (!state->logging || state->bus)
+	return RPMRC_OK;
+
+    if (lstat("/run/systemd/system/", &st) == 0) {
+        if (S_ISDIR(st.st_mode)) {
+	    // initialise the error value
```

Comment:
Please use `/* ... */` for comments, // is not in the rpm coding style even if the compiler permits it.